linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -2523,26 +2496,23 @@ private[optimizer] abstract class OptimizerCore(config: CommonPhaseConfig) {
                 Transient(ObjectClassName(expr))))
 
           case _ =>
-            defaultApply(StringClassType)
+            default
         }
 
       // java.lang.reflect.Array
 
       case ArrayNewInstance =>
-        newArgs.head match {
-          case ClassOf(elementTypeRef) =>
+        targs.head match {
+          case PreTransTree(ClassOf(elementTypeRef), _) =>
             val arrayTypeRef = ArrayTypeRef.of(elementTypeRef)
             contTree(NewArray(arrayTypeRef, List(newArgs.tail.head)))
```

Comment:
Even here, we're only accessing `newArgs.tail` but that uselessly compute `finishTransformExpr` for `targs.head`. (although it's not an issue in this case since we know it's a `PreTransTree` anyway).